src/main/java/com/xjeffrose/chicago/ChicagoObjectDecoder.java

```diff
@@ -72,44 +47,77 @@ private ChicagoMessage _decode(ByteBuf msg) {
     final byte[] valSize = new byte[4];
 
     // Determine the message ID
+    if (msg.readableBytes() < id.length) {
+      return null;
+    }
     msg.readBytes(id, 0, id.length);
 
     // Determine the operation type
+    if (msg.readableBytes() < op.length) {
+      msg.readerIndex(startingPosition);
+      return null;
+    }
     msg.readBytes(op, 0, op.length);
 
     // Get the colFam Length
+    if (msg.readableBytes() < colFamSize.length) {
+      msg.readerIndex(startingPosition);
+      return null;
+    }
     msg.readBytes(colFamSize, 0, colFamSize.length);
     final int colFamLength = Ints.fromByteArray(colFamSize);
     final byte[] colFam = new byte[colFamLength];
 
     // Get the Col Fam
+    if (msg.readableBytes() < colFam.length) {
+      msg.readerIndex(startingPosition);
+      return null;
```

Comment:
Even if we have 50 bytes of data we won't know how long the actual message is.  So we might end up trying to read more than what is available at this time, which would throw an exception.  These `readableBytes` checks prevent us from reading too much, and allow us to abort the message read by resetting the readerIndex and returning null to break out of the caller loop.